rest-api/src/main/java/com/cloudbees/workflow/rest/external/StageNodeExt.java

```diff
@@ -55,23 +57,7 @@ public void setStageFlowNodes(List<AtomFlowNodeExt> stageFlowNodes) {
     }
 
     public static boolean isStageNode(FlowNode node) {
-        return (node.getAction(StageAction.class) != null);
-    }
-
-    public static StageNodeExt create(FlowNode node) {
-        StageNodeExt stageNodeExt = new StageNodeExt();
-
-        stageNodeExt.addBasicNodeData(node);
-
-        // Use the last node in the stage to configure the stage status.
-        FlowNode stageEndNode = FlowNodeUtil.getStageEndNode(node);
-        if (NotExecutedNodeAction.isExecuted(stageEndNode)) {
-            stageNodeExt.setStatus(StatusExt.valueOf(stageEndNode.getError()));
-        } else {
-            stageNodeExt.setStatus(StatusExt.NOT_EXECUTED);
-        }
-
-        return stageNodeExt;
+        return (node.getAction(LabelAction.class) != null && node.getAction(ThreadNameAction.class) == null);
```

Comment:
I would add something like <code>TODO: is it really false-positive-safe?</code> to the code. It seems to be fragile, but I would not expect an issue unless somebody wants to add it intentionally